diff --git a/libavformat/flvdec.c b/libavformat/flvdec.c
index 22a9b9e4a7..0cc0a08fe2 100644
--- a/libavformat/flvdec.c
+++ b/libavformat/flvdec.c
@@ -73,6 +73,7 @@ typedef struct FLVContext {
     int trust_metadata;   ///< configure streams according onMetaData
     int trust_datasize;   ///< trust data size of FLVTag
     int dump_full_metadata;   ///< Dump full metadata of the onMetadata
+    int drop_negative_cts;///< drop frames if cts is negative
     int wrong_dts;        ///< wrong dts due to negative cts
     uint8_t *new_extradata[FLV_STREAM_TYPE_NB];
     int new_extradata_size[FLV_STREAM_TYPE_NB];
@@ -1468,10 +1469,15 @@ retry_duration:
             int32_t cts = (avio_rb24(s->pb) + 0xff800000) ^ 0xff800000;
             pts = av_sat_add64(dts, cts);
             if (cts < 0) { // dts might be wrong
-                if (!flv->wrong_dts)
-                    av_log(s, AV_LOG_WARNING,
-                        "Negative cts, previous timestamps might be wrong.\n");
-                flv->wrong_dts = 1;
+                if (flv->drop_negative_cts) {
+                    av_log(s, AV_LOG_WARNING,"Negative cts, frames will be dropped.\n");
+                    dts = pts = AV_NOPTS_VALUE;
+                } else {
+                    if (!flv->wrong_dts)
+                        av_log(s, AV_LOG_WARNING,
+                               "Negative cts, previous timestamps might be wrong.\n");
+                    flv->wrong_dts = 1;
+                }
             } else if (FFABS(dts - pts) > 1000*60*15) {
                 av_log(s, AV_LOG_WARNING,
                        "invalid timestamps %"PRId64" %"PRId64"\n", dts, pts);
@@ -1576,6 +1582,7 @@ static const AVOption options[] = {
     { "flv_metadata", "Allocate streams according to the onMetaData array", OFFSET(trust_metadata), AV_OPT_TYPE_BOOL, { .i64 = 0 }, 0, 1, VD },
     { "flv_full_metadata", "Dump full metadata of the onMetadata", OFFSET(dump_full_metadata), AV_OPT_TYPE_BOOL, { .i64 = 0 }, 0, 1, VD },
     { "flv_ignore_prevtag", "Ignore the Size of previous tag", OFFSET(trust_datasize), AV_OPT_TYPE_BOOL, { .i64 = 0 }, 0, 1, VD },
+    { "flv_drop_negative_cts", "Drop frames with negative composition timestamp", OFFSET(drop_negative_cts), AV_OPT_TYPE_BOOL, { .i64 = 0 }, 0, 1, VD },
     { "missing_streams", "", OFFSET(missing_streams), AV_OPT_TYPE_INT, { .i64 = 0 }, 0, 0xFF, VD | AV_OPT_FLAG_EXPORT | AV_OPT_FLAG_READONLY },
     { NULL }
 };
